Sources/VLCPlaybackController.m

```diff
@@ -1165,46 +1077,40 @@ - (void)setNeedsMetadataUpdate
 }
 
 #if TARGET_OS_IOS
-- (void)_recoverLastPlaybackStateOfItem:(MLFile *)item
-{
-    if (item) {
-        CGFloat lastPosition = .0;
-        NSInteger duration = 0;
-
-        if (item.lastPosition) {
-            lastPosition = item.lastPosition.floatValue;
-        }
-        
-        duration = item.duration.intValue;
-
-        if (lastPosition < .95 && _mediaPlayer.position < lastPosition) {
-            NSInteger continuePlayback;
-            if ([item isAlbumTrack] || [item isSupportedAudioFile])
-                continuePlayback = [[[NSUserDefaults standardUserDefaults] objectForKey:kVLCSettingContinueAudioPlayback] integerValue];
-            else
-                continuePlayback = [[[NSUserDefaults standardUserDefaults] objectForKey:kVLCSettingContinuePlayback] integerValue];
-
-            if (continuePlayback == 1) {
-                [self setPlaybackPosition:lastPosition];
-            } else if (continuePlayback == 0) {
-                NSArray<VLCAlertButton *> *buttonsAction = @[[[VLCAlertButton alloc] initWithTitle: NSLocalizedString(@"BUTTON_CANCEL", nil)
-                                                                                             style: UIAlertActionStyleCancel
-                                                                                            action: nil],
-                                                             [[VLCAlertButton alloc] initWithTitle: NSLocalizedString(@"BUTTON_CONTINUE", nil)
-                                                                                            action: ^(UIAlertAction *action) {
-                                                                                                [self setPlaybackPosition:lastPosition];
-                                                                                            }]
-                                                             ];
-                UIViewController *presentingVC = [UIApplication sharedApplication].delegate.window.rootViewController;
-                presentingVC = presentingVC.presentedViewController ?: presentingVC;
-                [VLCAlertViewController alertViewManagerWithTitle:NSLocalizedString(@"CONTINUE_PLAYBACK", nil)
-                                                     errorMessage:[NSString stringWithFormat:NSLocalizedString(@"CONTINUE_PLAYBACK_LONG", nil), item.title]
-                                                   viewController:presentingVC
-                                                    buttonsAction:buttonsAction];
+- (void)_recoverLastPlaybackState
+{
+    VLCMLMedia *media = [_delegate mediaForPlayingMedia:_mediaPlayer.media];
+    if (!media) return;
+
+    CGFloat lastPosition = media.progress;
+    if (_mediaPlayer.position < lastPosition) {
```

Comment:
note that I removed the .95 percent check knowingly, when you have tv shows it tended to not restart even though you weren't finished